mitmproxy/protocol/http.py

```diff
@@ -170,13 +172,33 @@ def __call__(self):
             try:
                 flow = models.HTTPFlow(self.client_conn, self.server_conn, live=self)
                 flow.request = request
+
                 # set upstream auth
                 if self.mode == "upstream" and self.config.upstream_auth is not None:
                     flow.request.headers["Proxy-Authorization"] = self.config.upstream_auth
                 self.process_request_hook(flow)
 
+                # WebSockets
+                if websockets.check_handshake(request.headers):
```

Comment:
What if I want to intercept and modify the websockets handshake?
Moving it down basically only gives me a completed handshake - without the possibility to do anything afterwards?